### PR TITLE
accept scala classes as params types

### PR DESCRIPTION
**Pull request by chandu0101**
done! .I used value classes for test, cause its easy :p , let me know if you need a test case for non value class.

### CODE REVIEW

**Comment by ramnivas** [comment]
"let me know if you need a test case for non value class." That will be nice and will ensure continued compliance as we make other changes. If you do add a test, I'll prefer it to be separate test so that failures in future are better diagnosable.

File: src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala [hunk 1]
```diff
@@ -39,12 +43,22 @@ class ReactBridgeComponentTest extends FunSuite {
...
     assert(mounted.getDOMNode().querySelector("#props").textContent === "{one->1,two->2,foo->bar}")
   }
 
+  test("object properties") {
+    case class TestComponent(id: js.UndefOr[String] = js.undefined, className: js.UndefOr[String] = js.undefined,
+                             ref: js.UndefOr[String] = js.undefined, key: js.UndefOr[Any] = js.undefined,
+                             props: js.UndefOr[NameType] = js.undefined) extends ReactBridgeComponent
+
+    val testComponent: ReactElement = TestComponent(props = new NameType("dude"))()
+    println(React.renderToStaticMarkup(testComponent))
```
Comment (ramnivas) [inline src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala]: Can you take out this println?

Comment (chandu0101) [inline src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala]: ah i removed from map test case :( , do you want me to remove all printlns ?

Comment (ramnivas) [inline src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala]: That will be nice :-)

Comment (chandu0101) [inline src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala]: done ,please review it :)

**Comment by ramnivas** [comment]
Merged via 67f80b0a623ec8a3c5a550e174c2a2b0418b92e7

**Pull request closed without merging**